core-web/libs/sdk/analytics/src/lib/core/shared/models/event.model.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Event models for DotCMS Analytics
+ * Contains interfaces for different types of analytics events
+ */
+
+import { DotCMSDeviceData, DotCMSPageData, DotCMSUtmData } from './data.model';
+
+import {
+    CustomEventType,
+    DotCMSEventType,
+    DotCMSPredefinedEventType
+} from '../constants/dot-content-analytics.constants';
+
+/**
+ * Base structure for all analytics events.
+ * All events share this common structure.
+ */
+export interface DotCMSEventBase {
+    /** The type of event being tracked */
+    event_type: DotCMSEventType;
+    /** Local timestamp when the event occurred */
+    local_time: string;
+}
+
+/**
+ * Pageview-specific analytics event structure.
+ * Contains data specific to page view tracking.
+ */
+export interface DotCMSPageViewEvent extends DotCMSEventBase {
+    event_type: typeof DotCMSPredefinedEventType.PAGEVIEW;
+    /** Pageview-specific event data with structured format */
+    data: {
+        /** Page data associated with the event */
+        page: DotCMSPageData;
+        /** Device and browser information */
+        device: DotCMSDeviceData;
+        /** UTM parameters for campaign tracking (optional) */
+        utm?: DotCMSUtmData;
+        /** Custom data associated with the event */
+        custom?: Record<string, unknown>;
+    };
+}
+
+/**
+ * Custom event structure.
+ * Contains data for user-defined custom events.
+ */
+export interface DotCMSCustomEvent extends DotCMSEventBase {
+    /** The type of event being tracked (any string except structured events) */
+    event_type: CustomEventType;
```

Comment:
```suggestion
    event_type: event_type: typeof DotCMSPredefinedEventType.CUSTOM;;
```

Reply:
the last one
```
event_type: typeof DotCMSPredefinedEventType.CUSTOM;;
```

The event_type is whatever the client wants, is any string except `pageview` (by now, I will add later content_impression and so on.)

```
export type DotCMSCustomEventType = Exclude<string, DotCMSPredefinedEventType>;
```